worker/src/ee/cloudSpendAlerts/handleCloudSpendAlertJob.ts

```diff
@@ -0,0 +1,258 @@
+import { parseDbOrg, Role } from "@langfuse/shared";
+import { prisma } from "@langfuse/shared/src/db";
+import Stripe from "stripe";
+import { env } from "../../env";
+import { logger } from "@langfuse/shared/src/server";
+import { recordIncrement, traceException } from "@langfuse/shared/src/server";
+import { Job } from "bullmq";
+import { backOff } from "exponential-backoff";
+import { sendCloudSpendAlertEmail } from "@langfuse/shared/src/server";
+
+export const handleCloudSpendAlertJob = async (job: Job<{ orgId: string }>) => {
+  const { orgId } = job.data;
+
+  logger.info(`[CLOUD SPEND ALERTS] Processing org ${orgId}`);
+
+  if (!env.STRIPE_SECRET_KEY) {
+    logger.warn("[CLOUD SPEND ALERTS] Stripe secret key not found");
+    throw new Error("Stripe secret key not found");
+  }
+
+  // Fetch organization with spend alerts
+  const orgData = await prisma.organization.findFirst({
+    where: {
+      id: orgId,
+    },
+    include: {
+      cloudSpendAlerts: true,
+    },
+  });
+
+  if (!orgData) {
+    logger.error(`[CLOUD SPEND ALERTS] Organization ${orgId} not found`);
+    return;
+  }
+
+  const org = {
+    ...parseDbOrg(orgData),
+    spendAlerts: orgData.cloudSpendAlerts,
+  };
+
+  // Check if org has spend alerts configured
+  if (org.spendAlerts.length === 0) {
+    logger.info(`[CLOUD SPEND ALERTS] No spend alerts for org ${orgId}`);
+    return;
+  }
+
+  if (org.cloudConfig?.plan === "Hobby") {
+    // handle case where user has downgraded to hobby
+    logger.info(
+      `[CLOUD SPEND ALERTS] Org ${orgId} not entitled to spend alerts (plan: Hobby"})`,
+    );
+    return;
+  }
```

Comment:
Should we remove the spendAlert in this case from the table?

Reply:
I would keep them – if users accidentally downgraded and want to re-upgrade their old spend alerts will not be gone.

I think the cost in terms of practical additional compute is negligable.